pkg/compactor/compactor_test.go

```diff
@@ -1220,6 +1220,151 @@ func TestMultitenantCompactor_ShouldCompactOnlyUsersOwnedByTheInstanceOnSharding
 	}
 }
 
+func TestMultitenantCompactor_ShouldFailWithInvalidTSDBCompactOutput(t *testing.T) {
```

Comment:
Now that you moved the logic to a dedicated function `verifyCompactedBlocksTimeRanges()` I think will be easier to unit test all cases directly running that function. I would keep this test as an higher level test, but I would just run 1 single smoke test (the case the verification fail) while all different scenarios may be easier tested unit testing `verifyCompactedBlocksTimeRanges()`.